src/utils/components/DiskModal/DiskModal.tsx

```diff
@@ -87,11 +88,12 @@ const DiskModal: React.FC<DiskModalProps> = ({
         );
       }
       if (diskState.diskSource === sourceTypes.UPLOAD) {
-        return getPersistentVolumeClaimHotplugPromise(
-          vmObj,
-          `${vm?.metadata?.name}-${diskState.diskName}`,
-          resultDisk,
-        );
+        const isNameTemplateParameter = /[${}]/.test(vm?.metadata?.name);
```

Comment:
Are you sure this kind of check is gonna work well everywhere we use DiskModal? I can see we deal with disks when dealing with VMs and also Templates.. And when you get here data of a template that has been already processed, there's no any chars like `${}` in the name even when the name was generated according to the `NAME` param.

Buut... maybe this is not a problem? I am not sure because this PR lacks more info about the problem in its description. Maybe the problem is just we don't wanna get `${}` chars into the `pvcName`?

Reply:
As you can see from the bugzilla ticket, there is also a screenshot of the problem.
The description is explanatory to me. The Diskmodal is trying to create a disk with `${NAME}`. `$`, `{` and `}` cannot be used as names so whatever name it is, if contains those characters, the name contains a parameter. 

The template at this point is not processed.

Reply:
Thanks for the explanation, Ugo. So you are sure it's gonna work well for both VM and Templates Disks pages?